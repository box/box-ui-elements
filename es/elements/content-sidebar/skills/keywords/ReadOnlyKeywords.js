function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/**
 * 
 * @file Read Only Keywords Card component
 * @author Box
 */

import * as React from 'react';
import PillCloud from '../../../../components/pill-cloud/PillCloud';
import { SKILLS_TARGETS, INTERACTION_TARGET } from '../../../common/interactionTargets';
import Timeline from '../timeline';
import getPills from './keywordUtils';
import './ReadOnlyKeywords.scss';
class ReadOnlyselecteds extends React.PureComponent {
  constructor(...args) {
    super(...args);
    _defineProperty(this, "state", {
      selectedIndex: -1
    });
    /**
     * Shows the time line by selecting the keyword
     *
     * @private
     * @param {Object} pill - keyword
     * @return {void}
     */
    _defineProperty(this, "onSelect", pill => {
      const {
        selectedIndex
      } = this.state;
      const newIndex = pill.value;
      this.setState({
        selectedIndex: selectedIndex === newIndex ? -1 : newIndex
      });
    });
  }
  /**
   * Renders the keywords
   *
   * @private
   * @return {void}
   */
  render() {
    const {
      keywords,
      getViewer,
      duration
    } = this.props;
    const {
      selectedIndex
    } = this.state;
    const options = getPills(keywords);
    const selected = keywords[selectedIndex];
    const pillCloudProps = selected ? {
      selectedOptions: [options[selectedIndex]]
    } : {};
    return /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(PillCloud, _extends({
      onSelect: this.onSelect,
      options: options
    }, pillCloudProps, {
      buttonProps: {
        [INTERACTION_TARGET]: SKILLS_TARGETS.KEYWORDS.SELECT
      }
    })), !!selected && Array.isArray(selected.appears) && selected.appears.length > 0 && /*#__PURE__*/React.createElement(Timeline, {
      duration: duration,
      getViewer: getViewer,
      interactionTarget: SKILLS_TARGETS.KEYWORDS.TIMELINE,
      text: selected.text,
      timeslices: selected.appears
    }));
  }
}
export default ReadOnlyselecteds;
//# sourceMappingURL=ReadOnlyKeywords.js.map