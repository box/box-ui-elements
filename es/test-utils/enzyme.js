function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _defineProperty(e, r, t) { return (r = _toPropertyKey(r)) in e ? Object.defineProperty(e, r, { value: t, enumerable: !0, configurable: !0, writable: !0 }) : e[r] = t, e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == typeof i ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != typeof t || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != typeof i) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); }
/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({
  children
}) => {
  return /*#__PURE__*/React.createElement(ThemeProvider, {
    theme: defaultTheme
  }, children);
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, _objectSpread({
  wrappingComponent: Wrappers
}, options));

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
//# sourceMappingURL=enzyme.js.map