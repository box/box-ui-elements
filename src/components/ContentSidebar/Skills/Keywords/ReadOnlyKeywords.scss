@import '../../../variables';

.be-keywords {
    .pill-cloud-container {
        border: 0 none;
        overflow: hidden;
        padding: 0;

        .btn.pill.pill-cloud-button {
            border: 1px solid $see-see;
            color: inherit;
            margin: 0 5px 5px 0;

            &:hover,
            &.is-selected {
                background-color: $blue;
                border-color: $blue;
                color: $white;
            }
        }
    }
}
