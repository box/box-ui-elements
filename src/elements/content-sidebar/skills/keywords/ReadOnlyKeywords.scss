@import '../../../common/variables';

.be-keywords {
    .bdl-PillCloud,
    .pill-cloud-container {
        padding: 0;
        overflow: hidden;
        border: 0 none;

        .btn.pill.pill-cloud-button {
            margin: 0 5px 5px 0;
            color: inherit;
            border: 1px solid $bdl-gray-20;

            &:hover,
            &.is-selected {
                color: $white;
                background-color: $blue;
                border-color: $blue;
            }
        }
    }
}
