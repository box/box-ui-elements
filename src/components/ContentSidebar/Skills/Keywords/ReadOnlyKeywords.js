/**
 * @flow
 * @file Read Only Keywords Card component
 * @author Box
 */

import * as React from 'react';
import PillCloud from 'box-react-ui/lib/components/pill-cloud/PillCloud';
import Timeline from '../Timeline';
import getPills from './keywordUtils';
import {
    SKILLS_TARGETS,
    INTERACTION_TARGET,
} from '../../../../interactionTargets';
import type { Pill, Pills } from './flowTypes';
import './ReadOnlyKeywords.scss';

type Props = {
    duration?: number,
    keywords: Array<SkillCardEntry>,
    getViewer?: Function,
};

type State = {
    selectedIndex: number,
};

class ReadOnlyselecteds extends React.PureComponent<Props, State> {
    props: Props;

    state: State = {
        selectedIndex: -1,
    };

    /**
     * Shows the time line by selecting the keyword
     *
     * @private
     * @param {Object} pill - keyword
     * @return {void}
     */
    onSelect = (pill: Pill) => {
        const { selectedIndex }: State = this.state;
        const newIndex = pill.value;
        this.setState({
            selectedIndex: selectedIndex === newIndex ? -1 : newIndex,
        });
    };

    /**
     * Renders the keywords
     *
     * @private
     * @return {void}
     */
    render() {
        const { keywords, getViewer, duration }: Props = this.props;
        const { selectedIndex }: State = this.state;
        const options: Pills = getPills(keywords);
        const selected = keywords[selectedIndex];
        const pillCloudProps = selected
            ? { selectedOptions: [options[selectedIndex]] }
            : {};

        return (
            <React.Fragment>
                <PillCloud
                    options={options}
                    onSelect={this.onSelect}
                    {...pillCloudProps}
                    buttonProps={{
                        [INTERACTION_TARGET]: SKILLS_TARGETS.KEYWORDS.SELECT,
                    }}
                />
                {!!selected &&
                    Array.isArray(selected.appears) &&
                    selected.appears.length > 0 && (
                        <Timeline
                            text={selected.text}
                            timeslices={selected.appears}
                            duration={duration}
                            getViewer={getViewer}
                            interactionTarget={SKILLS_TARGETS.KEYWORDS.TIMELINE}
                        />
                    )}
            </React.Fragment>
        );
    }
}

export default ReadOnlyselecteds;
