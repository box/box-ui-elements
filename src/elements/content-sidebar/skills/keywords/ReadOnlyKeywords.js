/**
 * @flow
 * @file Read Only Keywords Card component
 * @author Box
 */

import * as React from 'react';
import PillCloud from '../../../../components/pill-cloud/PillCloud';
import { SKILLS_TARGETS, INTERACTION_TARGET } from '../../../common/interactionTargets';
import Timeline from '../timeline';
import getPills from './keywordUtils';
import type { Pill, Pills } from './flowTypes';
import type { SkillCardEntry } from '../../../../common/types/skills';
import './ReadOnlyKeywords.scss';

type Props = {
    duration?: number,
    getViewer?: Function,
    keywords: Array<SkillCardEntry>,
};

type State = {
    selectedIndex: number,
};

class ReadOnlyselecteds extends React.PureComponent<Props, State> {
    props: Props;

    state: State = {
        selectedIndex: -1,
    };

    /**
     * Shows the time line by selecting the keyword
     *
     * @private
     * @param {Object} pill - keyword
     * @return {void}
     */
    onSelect = (pill: Pill) => {
        const { selectedIndex }: State = this.state;
        const newIndex: number = ((pill.value: any): number);
        this.setState({
            selectedIndex: selectedIndex === newIndex ? -1 : newIndex,
        });
    };

    /**
     * Renders the keywords
     *
     * @private
     * @return {void}
     */
    render() {
        const { keywords, getViewer, duration }: Props = this.props;
        const { selectedIndex }: State = this.state;
        const options: Pills = getPills(keywords);
        const selected = keywords[selectedIndex];
        const pillCloudProps = selected ? { selectedOptions: [options[selectedIndex]] } : {};

        return (
            <>
                <PillCloud
                    onSelect={this.onSelect}
                    options={options}
                    {...pillCloudProps}
                    buttonProps={{
                        [INTERACTION_TARGET]: SKILLS_TARGETS.KEYWORDS.SELECT,
                    }}
                />
                {!!selected && Array.isArray(selected.appears) && selected.appears.length > 0 && (
                    <Timeline
                        duration={duration}
                        getViewer={getViewer}
                        interactionTarget={SKILLS_TARGETS.KEYWORDS.TIMELINE}
                        text={selected.text}
                        timeslices={selected.appears}
                    />
                )}
            </>
        );
    }
}

export default ReadOnlyselecteds;
