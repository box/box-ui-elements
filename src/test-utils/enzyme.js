/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';
import { mount as baseMount, shallow as baseShallow } from 'enzyme';

// Global providers that need setup
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../styles/theme';

// Wrap this around every component so they can use app-wide context.
// Prevents breaking due to missing providers.
// See https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/getWrappingComponent.md
/* eslint-disable react/prop-types */
const Wrappers = ({ children }) => {
    return <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>;
};

/**
 * mount() from Enzyme but with required app-wide context providers included by default
 */
const mountConnected = (element, options = {}) => baseMount(element, { wrappingComponent: Wrappers, ...options });

/**
 * shallow() from Enzyme but with required app-wide context providers included by default.
 */
const shallowConnected = (element, options = {}) => baseShallow(element, { wrappingComponent: Wrappers, ...options });

// Re-export everything from Enzyme
export * from 'enzyme';
// Export wrapped methods
export { mountConnected, shallowConnected };
