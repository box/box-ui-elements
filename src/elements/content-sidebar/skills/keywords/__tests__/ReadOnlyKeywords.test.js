import * as React from 'react';
import { shallow } from 'enzyme';
import ReadOnlyKeywords from '../ReadOnlyKeywords';

describe('elements/content-sidebar/Skills/Keywords/ReadOnlyKeywords', () => {
    test('should correctly render with no keyword selected', () => {
        const props = {
            keywords: [
                { text: 'foo', appears: [{ start: 1 }] },
                { text: 'bar', appears: [{ start: 5 }] },
            ],
        };

        const wrapper = shallow(<ReadOnlyKeywords {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render timeline with keyword selected', () => {
        const props = {
            keywords: [
                { text: 'foo', appears: [{ start: 1 }] },
                { text: 'bar', appears: [{ start: 5 }] },
            ],
        };

        const wrapper = shallow(<ReadOnlyKeywords {...props} />);
        wrapper.setState({ selected: { text: 'foo', value: 1 } });

        expect(wrapper).toMatchSnapshot();
    });
});
