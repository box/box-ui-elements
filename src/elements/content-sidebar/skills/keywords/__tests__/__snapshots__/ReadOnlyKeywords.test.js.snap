// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements/content-sidebar/Skills/Keywords/ReadOnlyKeywords should correctly render timeline with keyword selected 1`] = `
<Fragment>
  <PillCloud
    buttonProps={
      {
        "data-resin-target": "skill-keywordselect",
      }
    }
    onSelect={[Function]}
    options={
      [
        {
          "displayText": "foo",
          "value": 0,
        },
        {
          "displayText": "bar",
          "value": 1,
        },
      ]
    }
  />
</Fragment>
`;

exports[`elements/content-sidebar/Skills/Keywords/ReadOnlyKeywords should correctly render with no keyword selected 1`] = `
<Fragment>
  <PillCloud
    buttonProps={
      {
        "data-resin-target": "skill-keywordselect",
      }
    }
    onSelect={[Function]}
    options={
      [
        {
          "displayText": "foo",
          "value": 0,
        },
        {
          "displayText": "bar",
          "value": 1,
        },
      ]
    }
  />
</Fragment>
`;
